services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/reroute/actions/ValidateFlowAction.java

```diff
@@ -66,29 +67,31 @@ public ValidateFlowAction(PersistenceManager persistenceManager, FlowOperationsD
                 new HashSet<>(Optional.ofNullable(context.getPathsToReroute()).orElse(emptySet()));
         dashboardLogger.onFlowPathReroute(flowId, pathsToReroute, context.isForceReroute());
 
+        final boolean forceDefaultEncapsulation = featureTogglesRepository.find()
+                .map(FeatureToggles::getFlowsRerouteUsingDefaultEncapType)
+                .orElseGet(FeatureToggles.DEFAULTS::getFlowsRerouteUsingDefaultEncapType);
+
         Flow flow = persistenceManager.getTransactionManager().doInTransaction(() -> {
-            Flow foundFlow = getFlow(flowId, FetchStrategy.NO_RELATIONS);
+            Flow foundFlow = getFlow(flowId, FetchStrategy.DIRECT_RELATIONS);
```

Comment:
DIRECT_RELATIONS is not enough. Flow has relation to path. Path has relation to src/dstSwitch. 
in `new Flow(foundFlow)` constructor you use `flow.path.src/dstSwitch` field. It will be null with `DIRECT_RELATIONS` strategy